Spec/RealtimeClient.swift

```diff
@@ -1328,6 +1328,56 @@ class RealtimeClient: QuickSpec {
                 expect(result).to(equal(expectedOrder))
             }
 
+            // Possible issue of https://github.com/ably/ably-ios/issues/640
+            xit("should unlock queue when the timer ends") {
+                let options = ARTClientOptions(key: "xxxx:xxxx")
+                options.autoConnect = false
+                let client = ARTRealtime(options: options)
+                let channel = client.channels.get("foo")
+
+                waitUntil(timeout: testTimeout) { done in
+                    // Schedule a timer in the `internalDispatchQueue`
+                    let listener = client.internalEventEmitter.once { _ in }
+                    listener.setTimer(0.2, onTimeout: {
+                        // another lock
+                        channel.unsubscribe()
+                        done()
+                    })
+                    listener.startTimer()
+
+                    // lock
+                    channel.unsubscribe()
+                }
+            }
+
+            class AblyManager {
+                static let sharedClient = ARTRealtime(options: { $0.autoConnect = false; return $0 }(ARTClientOptions(key: "xxxx:xxxx")))
+            }
+
+            // Possible issue of https://github.com/ably/ably-ios/issues/640
+            xit("should dispatch in user queue") {
+                class Foo {
+                    let channel = "foo"
+                    init() {
+                        AblyManager.sharedClient.channels.get(channel).subscribe { _ in
+                            // keep reference
+                            self.update()
+                        }
+                    }
+                    deinit {
+                        close()
+                    }
+                    func update() {
+                    }
+                    func close() {
+                        AblyManager.sharedClient.channels.get(channel).unsubscribe()
+                    }
+                }
+
+                let foo = Foo()
+                foo.close()
```

Comment:
Hmm, does this really reproduce the issue? I think if you call `close` here, then it will remove the listener you add at `init` but since you still have a reference to the created object its `deinit` won't be called, and when you drop the reference just after, since you already removed the listener, its `deinit`'s `unsubscribe` will just do nothing.

I think `deinit` should just call `unsubscribe` directly; we don't need `close`.

Please confirm this reproduces the issue by commenting out the `dispatch_async` in `removeListener` and making sure this test then fails, and stops failing if you put it back.

Reply:
> Hmm, does this really reproduce the issue?

I think so but it always pass. I'll change the code with your points.

Reply:
It always passes d0b8aeb

Reply:
Should we ask the client to test using the branch?

Reply:
@ricardopereira Yes, but we also need to have a test that actually reproduces the problem. Maybe instead of `self` we can reference a different object, then make sure we drop the other reference to it before dropping the reference to `foo`?